### PR TITLE
unit_tests: Add RPC unit test

### PR DESCRIPTION
**Pull request by sam-golioth**
Adds unit tests for the RPC module.

Also adds a set of fake functions for golioth_coap_client and golioth_time.

Also cleans up the Unit Test CMakeLists.txt and adds a convenience function for declaring unit tests.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -8,25 +8,61 @@ set(CMAKE_C_FLAGS "-fprofile-arcs -ftest-coverage")
...
+    fakes/golioth_time_fake.c
+)
+target_include_directories(test_golioth_rpc PRIVATE
+    ${repo_root}/external/libcoap/include
+    ${repo_root}/port/esp_idf/libcoap/include
+    ${repo_root}/port/linux
 )
-target_link_libraries(${test_name} unity fff)
-add_test(NAME ${test_name} COMMAND "./${test_name}")
+target_link_libraries(test_golioth_rpc cjson)
```
Comment (mniestroj) [inline test/CMakeLists.txt:68]: There is no newline at the end of this file as well as some other files. Would be good to clean that up to remove noise in future commits. 

We use pre-commit to enforce that in CI for Zephyr SDK: https://github.com/golioth/golioth-zephyr-sdk/blob/main/.github/workflows/pre-commit.yml with https://github.com/golioth/golioth-zephyr-sdk/blob/main/.pre-commit-config.yaml configuration file. Maybe we should add such github action to this repo as well?

File: src/golioth_rpc.c [hunk 2]
```diff
@@ -127,6 +127,7 @@ static void on_rpc(
         }
     }
     if (!method_found) {
+        GLTH_LOGW(TAG, "Method %s not registered", rpc_method->valuestring);
```
Comment (mniestroj) [inline src/golioth_rpc.c:130]: This is not related to "Add RPC unit test", so would preferably split it to separate commit.

File: src/golioth_rpc.c [hunk 3]
```diff
@@ -17,7 +17,7 @@ LOG_TAG_DEFINE(golioth_rpc);
 // Request:
 //
 // {
-//      "id": "id_string,
+//      "id": "id_string",
```
Comment (mniestroj) [inline src/golioth_rpc.c:20]: This is not related to "Add RPC unit test", so would preferably split it to separate commit.

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -8,25 +8,61 @@ set(CMAKE_C_FLAGS "-fprofile-arcs -ftest-coverage")
...
+
+golioth_unit_test(test_golioth_rpc
+    test_golioth_rpc.c
+    fakes/golioth_coap_client_fake.c
+    fakes/golioth_time_fake.c
+)
+target_include_directories(test_golioth_rpc PRIVATE
+    ${repo_root}/external/libcoap/include
+    ${repo_root}/port/esp_idf/libcoap/include
+    ${repo_root}/port/linux
```
Comment (mniestroj) [inline test/CMakeLists.txt:66]: Why do we include both `esp_idf` as well as `linux` port include directories? Shouldn't single port directory be included?

File: test/CMakeLists.txt (hunk 1 above)
Comment (sam-golioth) [reply test/CMakeLists.txt:68]: I'll open another PR for adding the pre-commit hook

**Review by mniestroj — APPROVED** [review]
